core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -349,8 +350,10 @@ private DefaultFunctionResolver dayOfMonth() {
    * DAYOFWEEK(STRING/DATE/DATETIME/TIMESTAMP).
    * return the weekday index for date (1 = Sunday, 2 = Monday, …, 7 = Saturday).
    */
-  private DefaultFunctionResolver dayOfWeek() {
-    return define(BuiltinFunctionName.DAYOFWEEK.getName(),
+  private DefaultFunctionResolver dayOfWeek(FunctionName name) {
+    return define(name,
+        implWithProperties((functionProperties, arg) -> DateTimeFunction.dayOfWeekToday(
+            functionProperties.getQueryStartClock()), INTEGER, TIME),
```

Comment:
```suggestion
        implWithProperties(nullMissingHandlingWithProperties(functionProperties, arg) -> DateTimeFunction.dayOfWeekToday(
            functionProperties.getQueryStartClock())), INTEGER, TIME),
```

Reply:
And all related changes

Reply:
Works as is without needing to checkout files from other branches. Fixed in 8daa4baf8f4d26b229723884196bd1da23668861